### PR TITLE
Add Zmenu for regulatory features

### PR DESCRIPTION
**Pull request by azangru**
## Description
- Adds Zmenu for regulatory features
- Zmenu can download sequences of regulatory features
- Colour legend merged from https://github.com/Ensembl/ensembl-client/pull/988

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1954

## Deployment URL(s)
http://regulation.review.ensembl.org

### CODE REVIEW

File: src/content/app/genome-browser/components/zmenu/zmenus/RegulationZmenu.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const zmenuContent = payload.content[0] as ZmenuContentRegulation;
+  const featureMetadata = zmenuContent.metadata;
+
+  return {
+    id: featureMetadata.id,
+    regionName: featureMetadata.region_name,
+    featureType: featureMetadata.feature_type,
+    start: featureMetadata.start,
+    end: featureMetadata.end,
+    coreStart: featureMetadata.core_start,
```
Comment (jyothishnt) [inline src/content/app/genome-browser/components/zmenu/zmenus/RegulationZmenu.tsx:79]: Reminder to swap values of bound and core.

File: src/shared/components/instant-download/instant-download-fetch/fetchSequenceChecksums.ts [hunk 2]
```diff
@@ -305,3 +323,15 @@ export const fetchTranscriptSequenceMetadata = (
...
   ).then((data) => processTranscriptData(data.transcript));
 };
+
+export const fetchRegionSequenceChecksum = async (variables: {
+  genomeId: string;
+  regionName: string;
+}): Promise<string> => {
+  const data = await request<RegionSequenceChecksumQueryResult>(
+    '/api/thoas',
+    regioSequenceChecksumQuery,
```
Comment (jyothishnt) [inline src/shared/components/instant-download/instant-download-fetch/fetchSequenceChecksums.ts:333]:
```suggestion
    regionSequenceChecksumQuery,
```

File: src/shared/components/instant-download/instant-download-fetch/fetchSequenceChecksums.ts [hunk 3]
```diff
@@ -219,6 +227,16 @@ const transcriptChecksumsQuery = gql`
   ${transcriptQueryFragment}
 `;
 
+const regioSequenceChecksumQuery = gql`
```
Comment (jyothishnt) [inline src/shared/components/instant-download/instant-download-fetch/fetchSequenceChecksums.ts:230]:
```suggestion
const regionSequenceChecksumQuery = gql`
```

File: src/content/app/genome-browser/components/zmenu/zmenus/RegulationZmenu.tsx (hunk 1 above)
Comment (azangru) [reply src/content/app/genome-browser/components/zmenu/zmenus/RegulationZmenu.tsx:79]: This should work correctly now, I believe. The mistake was on the genome browser's side.

**Review by jyothishnt — APPROVED** [review]
Looks ok to me (with my limited knowledge) Happy to approve :-)

**Pull request merged**